File: Year-2023/LeetCode-September-challenge/Candy.js
var candy = function (ratings) {
    let n = ratings.length;
    let candies = new Array(n).fill(1);

    for (let i = 1; i < n; ++i)
        if (ratings[i] > ratings[i - 1] && candies[i] <= candies[i - 1])
            candies[i] = candies[i - 1] + 1;

    for (let i = n - 2; i >= 0; --i)
        if (ratings[i] > ratings[i + 1] && candies[i] <= candies[i + 1])
            candies[i] = candies[i + 1] + 1;

    let totalCandies = 0;
    for (let i = 0; i < n; ++i)
        totalCandies += candies[i];

    return totalCandies;
};

let ratings = [1,0,2];
console.log(candy(ratings));

File: Year-2023/LeetCode-September-challenge/Candy.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int candy(vector<int>& ratings) {
        int n = ratings.size();
        vector<int> candies(n, 1);

        for(int i = 1; i < n; ++i)
            if(ratings[i] > ratings[i - 1] && candies[i] <= candies[i - 1])
                candies[i] = candies[i - 1] + 1;

        for(int i = n - 2; i >= 0; --i)
            if(ratings[i] > ratings[i + 1] && candies[i] <= candies[i + 1])
                candies[i] = candies[i + 1] + 1;

        int totalCandies = 0;

        for(int i = 0; i < n; ++i) totalCandies += candies[i];

        return totalCandies;
    }
};

int main(){
    vector<int> ratings = {1,0,2};
    Solution *s = new Solution();
    cout<<s->candy(ratings)<<endl;
    return 0;
}